Source/JSONAPILinks.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  JSONAPILinks.swift
+//  Kakapo
+//
+//  Created by Joan Romano on 19/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+public enum JSONAPILink: CustomSerializable {
+    case Simple(value: String)
+    case Object(href: String, meta: Serializable)
```

Comment:
Which links where the one that were mutually exclusive? 
In this case you have two cases: one with href and the other with href and meta so an enumeration doesn't make sense